driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2116,6 +2116,18 @@ public PreparedStatement addPrepared(PreparedStatement stmt) {
             return stmt;
         }
 
+        public PreparedStatement replacePrepared(PreparedStatement stmt) {
+            PreparedStatement previous = preparedQueries.replace(stmt.getPreparedId().id, stmt);
+            if (previous != null) {
+                logger.debug("Replace already prepared query {}.", stmt.getQueryString());
+
+                // The one object in the cache will get GCed once it's not referenced by the client anymore since we use a weak reference.
+                // So we need to make sure that the instance we do return to the user is the one that is in the cache.
```

Comment:
This comment was copied from `addPrepared`, but I find it a bit confusing here, because client code doesn't use `previous` at all. In fact this method could be void.